src/plus/webviews/graph/graphWebview.ts

```diff
@@ -257,16 +279,48 @@ export class GraphWebview extends WebviewWithConfigBase<State> {
 	}
 }
 
-function formatCommits(commits: GitCommit[]): GitCommit[] {
-	return commits.map(({ sha, author, message, parents, committer }) => ({
-		sha: sha,
-		author: author,
-		message: message,
-		parents: parents,
-		committer: committer
+function formatCommits(commits: (GitCommit | GitStashCommit)[]): GitCommit[] {
+	return commits.map((commit: GitCommit) => ({
+		sha: commit.sha,
+		author: commit.author,
+		message: String(commit.message).length ? commit.message : commit.summary,
+		parents: commit.parents,
+		committer: commit.committer,
+		type: getCommitType(commit)
 	}));
 }
 
+// TODO: Move constant to a better home
+const enum CommitType {
+	CommitNode = 'commit-node',
+	StashNode = 'stash-node',
+}
```

Comment:
We could modify our `GraphContainer.d.ts` to to have those constants and then import them here.
We can do something similar as what we have done with `GraphZoneType`.

Then we will be able to remove this enum definition here (in 294 line).

Reply:
Addressed in commit: `Map to types in library`, but, I don't like the way I had to do it. I could not import from `gitkraken-components` in `graphWebView` without getting weird build errors that broke the build and the webview, so I had to do it in the wrapper and do a mapping there.

We will need to review/merge this PR first: https://github.com/gitkraken/GitKrakenComponents/pull/17

Reply:
We don't use enums for now in our library. Once we will convert our code to typescript we will be able to change that.
You don't need to do any mapping here. 

My idea was: 

- In `GraphContainer.d.ts` add following code instead of your changes:

```
import {
  commitNodeType,
  mergeConflictNodeType,
  mergeNodeType,
  stashNodeType,
  unsupportedRebaseWarningNodeType,
  workDirType
} from '../../domain/commit/CommitConstants';

...

export type WorkDirType = typeof mergeConflictNodeType
  | typeof unsupportedRebaseWarningNodeType
  | typeof workDirType;

export type CommitType = typeof commitNodeType
  | typeof mergeNodeType
  | typeof stashNodeType
  | WorkDirType;
```

- Create a new file `/src/domain/commit/CommitConstants.d.ts` with following code:
```
export const commitNodeType = 'commit-node';
export const mergeConflictNodeType = 'merge-conflict-node';
export const mergeNodeType = 'merge-node';
export const unsupportedRebaseWarningNodeType = 'unsupported-rebase-warning-node';
export const stashNodeType = 'stash-node';
export const workDirType = 'work-dir-changes';

```

- Then in `Gitlens` side, use directly the constant. For example in `graphWebView.ts`:
```
import { commitNodeType } from '@gitkraken/gitkraken-components/lib/domain/commit/CommitConstants';
...
if (GitCommitModel.isStash(commit)) {
   type = commitNodeType;
}
...
```